cmd/influxd/inspect/compact_series_file.go

```diff
@@ -0,0 +1,184 @@
+package inspect
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
+
+	"github.com/influxdata/influxdb/storage"
+	"github.com/influxdata/influxdb/tsdb"
+	"github.com/spf13/cobra"
+	"golang.org/x/sync/errgroup"
+)
+
+var compactSeriesFileFlags = struct {
+	// Standard input/output, overridden for testing.
+	Stderr io.Writer
+	Stdout io.Writer
+
+	// Data path options
+	SeriesFilePath string // optional. Defaults to <engine_path>/engine/_series
+	IndexPath      string // optional. Defaults to <engine_path>/engine/index
+
+	Concurrency int // optional. Defaults to GOMAXPROCS(0)
+}{
+	Stderr: os.Stderr,
+	Stdout: os.Stdout,
+}
+
+// NewCompactSeriesFileCommand returns a new instance of Command with default setting applied.
+func NewCompactSeriesFileCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "compact-series-file",
+		Short: "Compacts the series file to removed deleted series.",
+		Long:  `This command will compact the series file by removing deleted series.`,
+		RunE:  RunCompactSeriesFile,
+	}
+
+	defaultPath := filepath.Join(os.Getenv("HOME"), "/.influxdbv2/engine/")
+	defaultSFilePath := filepath.Join(defaultPath, storage.DefaultSeriesFileDirectoryName)
+	defaultIndexPath := filepath.Join(defaultPath, storage.DefaultIndexDirectoryName)
+
+	cmd.Flags().StringVar(&compactSeriesFileFlags.SeriesFilePath, "sfile-path", defaultSFilePath, "Path to the Series File directory. Defaults to "+defaultSFilePath)
+	cmd.Flags().StringVar(&compactSeriesFileFlags.IndexPath, "tsi-path", defaultIndexPath, "Path to the TSI index directory. Defaults to "+defaultIndexPath)
+
+	cmd.Flags().IntVar(&compactSeriesFileFlags.Concurrency, "concurrency", runtime.GOMAXPROCS(0), "Number of workers to dedicate to compaction. Defaults to GOMAXPROCS")
```

Comment:
Should this potentially max out at `8` because that's the largest number of series file partitions one could accumulate in 2.x.